  0%|          | 0/16 [00:00<?, ?it/s]  6%|6         | 1/16 [01:02<15:40, 62.67s/it] 12%|#2        | 2/16 [03:27<20:23, 87.41s/it] 19%|#8        | 3/16 [04:31<17:23, 80.23s/it] 25%|##5       | 4/16 [05:41<15:24, 77.08s/it] 31%|###1      | 5/16 [08:11<18:11, 99.23s/it] 38%|###7      | 6/16 [09:12<14:36, 87.61s/it] 44%|####3     | 7/16 [10:09<11:46, 78.53s/it] 50%|#####     | 8/16 [11:05<09:34, 71.77s/it] 56%|#####6    | 9/16 [12:11<08:10, 70.05s/it]  0%|          | 0/15 [00:00<?, ?it/s]  7%|6         | 1/15 [01:06<15:31, 66.52s/it] 13%|#3        | 2/15 [02:44<16:26, 75.85s/it] 20%|##        | 3/15 [04:07<15:37, 78.15s/it] 27%|##6       | 4/15 [05:41<15:09, 82.71s/it] 33%|###3      | 5/15 [07:04<13:48, 82.88s/it] 40%|####      | 6/15 [08:24<12:18, 82.09s/it] 47%|####6     | 7/15 [09:44<10:50, 81.33s/it] 53%|#####3    | 8/15 [11:02<09:22, 80.39s/it] 60%|######    | 9/15 [12:39<08:33, 85.56s/it]  0%|          | 0/11 [00:00<?, ?it/s]  9%|9         | 1/11 [01:24<14:01, 84.19s/it] 18%|#8        | 2/11 [02:37<12:07, 80.83s/it] 27%|##7       | 3/11 [03:50<10:28, 78.55s/it] 36%|###6      | 4/11 [04:55<08:40, 74.37s/it]./picks.sh: line 2: 58061 Terminated: 15          /Users/rohannarain/anaconda3/bin/python retrieve_data.py
./picks.sh: line 2: 59695 Terminated: 15          /Users/rohannarain/anaconda3/bin/python retrieve_data.py
./picks.sh: line 2: 86963 Terminated: 15          /Users/rohannarain/anaconda3/bin/python retrieve_data.py
2019-08-23 14:25:00.755 python[88140:1800557] _LSScheduleNotificationFunction() failed on line 649 in function id scheduleApplicationNotification(LSNotificationCode, NSWorkspaceNotificationCenter *)
2019-08-23 14:25:00.756 python[88139:1800556] _LSScheduleNotificationFunction() failed on line 649 in function id scheduleApplicationNotification(LSNotificationCode, NSWorkspaceNotificationCenter *)
2019-08-23 14:25:00.765 python[88138:1800553] _LSScheduleNotificationFunction() failed on line 649 in function id scheduleApplicationNotification(LSNotificationCode, NSWorkspaceNotificationCenter *)
  0%|          | 0/14 [00:00<?, ?it/s]  7%|7         | 1/14 [01:29<19:26, 89.70s/it] 14%|#4        | 2/14 [03:02<18:05, 90.49s/it] 21%|##1       | 3/14 [04:47<17:25, 95.02s/it] 29%|##8       | 4/14 [06:29<16:10, 97.05s/it] 36%|###5      | 5/14 [08:07<14:35, 97.30s/it] 43%|####2     | 6/14 [09:51<13:14, 99.31s/it] 50%|#####     | 7/14 [11:40<11:56, 102.36s/it] 57%|#####7    | 8/14 [13:39<10:43, 107.26s/it] 64%|######4   | 9/14 [15:11<08:33, 102.71s/it] 71%|#######1  | 10/14 [16:50<06:46, 101.54s/it]Traceback (most recent call last):
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 643, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/util/retry.py", line 334, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/Users/rohannarain/anaconda3/lib/python3.6/http/client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "retrieve_data.py", line 472, in <module>
    generate_hits_data()
  File "retrieve_data.py", line 408, in generate_hits_data
    player_id = get_player_id_from_name(player)
  File "retrieve_data.py", line 50, in get_player_id_from_name
    return statsapi.lookup_player(player_name)[0]['id']
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/statsapi/__init__.py", line 933, in lookup_player
    r = get('sports_players',params)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/statsapi/__init__.py", line 1421, in get
    r = requests.get(url)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/requests/adapters.py", line 473, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

Getting data...
Traceback (most recent call last):
  File "train_model.py", line 37, in <module>
    hits_test = pd.read_csv("data/player_stats/player_stats_{}.csv".format(today))
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/pandas/io/parsers.py", line 678, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/pandas/io/parsers.py", line 440, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/pandas/io/parsers.py", line 787, in __init__
    self._make_engine(self.engine)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/pandas/io/parsers.py", line 1014, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/Users/rohannarain/anaconda3/lib/python3.6/site-packages/pandas/io/parsers.py", line 1708, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 384, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 695, in pandas._libs.parsers.TextReader._setup_parser_source
FileNotFoundError: File b'data/player_stats/player_stats_08_24_2019.csv' does not exist
[master 5370cce] predictions 08/24/19
 Committer: Rohan Narain <rohannarain@rohans-mbp-3.attlocal.net>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly. Run the
following command and follow the instructions in your editor to edit
your configuration file:

    git config --global --edit

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 90 insertions(+), 1 deletion(-)
fatal: could not read Username for 'https://github.com': Device not configured
  0%|          | 0/15 [00:00<?, ?it/s]  7%|6         | 1/15 [02:55<40:54, 175.33s/it] 13%|#3        | 2/15 [04:21<32:11, 148.60s/it] 20%|##        | 3/15 [05:41<25:36, 128.03s/it] 27%|##6       | 4/15 [07:05<21:02, 114.82s/it] 33%|###3      | 5/15 [08:36<17:55, 107.53s/it] 40%|####      | 6/15 [10:01<15:07, 100.84s/it] 47%|####6     | 7/15 [11:42<13:26, 100.81s/it] 53%|#####3    | 8/15 [12:59<10:57, 93.89s/it]  60%|######    | 9/15 [14:35<09:27, 94.53s/it] 67%|######6   | 10/15 [15:47<07:17, 87.56s/it] 73%|#######3  | 11/15 [17:00<05:33, 83.35s/it] 80%|########  | 12/15 [18:40<04:24, 88.19s/it] 87%|########6 | 13/15 [19:54<02:47, 83.93s/it] 93%|#########3| 14/15 [21:11<01:21, 81.90s/it]100%|##########| 15/15 [22:33<00:00, 82.13s/it]
  0%|          | 0/15 [00:00<?, ?it/s]  7%|6         | 1/15 [00:43<10:13, 43.85s/it] 13%|#3        | 2/15 [02:02<11:45, 54.23s/it] 20%|##        | 3/15 [02:59<11:02, 55.19s/it] 27%|##6       | 4/15 [03:44<09:34, 52.18s/it] 33%|###3      | 5/15 [05:00<09:52, 59.25s/it] 40%|####      | 6/15 [06:13<09:29, 63.32s/it] 47%|####6     | 7/15 [07:33<09:07, 68.45s/it] 53%|#####3    | 8/15 [08:55<08:27, 72.49s/it] 60%|######    | 9/15 [09:46<06:35, 65.90s/it] 67%|######6   | 10/15 [10:34<05:03, 60.67s/it] 73%|#######3  | 11/15 [11:53<04:24, 66.07s/it] 80%|########  | 12/15 [12:39<02:59, 59.99s/it] 87%|########6 | 13/15 [13:33<01:56, 58.30s/it] 93%|#########3| 14/15 [14:49<01:03, 63.53s/it]100%|##########| 15/15 [15:53<00:00, 63.67s/it]
Finished generating file: data/player_stats/player_stats_08_24_2019.csv
Finished generating file: data/player_stats/player_stats_08_25_2019.csv
Results for 08/24/2019 generated
Getting data...
Data retrieved.
Training logistic regression...
Finished training!
Finding best hyperparameters for AdaBoost...
Training AdaBoost...
Finished training!
Finding best hyperparameters for random forests...
Training random forests...
Finished training!
Model performance summary on validation set: 
             Model  Accuracy  Precision    Recall  F1 Score
0          Logreg  0.783080   0.774869  0.721951  0.747475
1  Random Forests  0.789588   0.767327  0.756098  0.761671
2        AdaBoost  0.774403   0.761658  0.717073  0.738693
Predictions for today: 
                    Name                  Team  Hit Probability
0          Trevor Story      Colorado Rockies         0.843995
1          Amed Rosario         New York Mets         0.841046
2          Hunter Pence         Texas Rangers         0.813288
3      Michael Brantley        Houston Astros         0.813259
4          Yuli Gurriel        Houston Astros         0.807357
5  Nicholas Castellanos          Chicago Cubs         0.807016
6           Jose Altuve        Houston Astros         0.802195
7           Ketel Marte  Arizona Diamondbacks         0.798738
8           Kolten Wong   St. Louis Cardinals         0.792600
9        Austin Meadows        Tampa Bay Rays         0.791981
[master 8789762] predictions 08/25/19
 Committer: Rohan Narain <rohannarain@rohans-mbp-3.attlocal.net>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly. Run the
following command and follow the instructions in your editor to edit
your configuration file:

    git config --global --edit

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 6 files changed, 759 insertions(+), 363 deletions(-)
 create mode 100644 data/model_stats/performance_08_25_2019.csv
 create mode 100644 data/past_results/past_results_08_24_2019.csv
 rewrite data/player_stats/player_stats_08_24_2019.csv (89%)
 create mode 100644 data/player_stats/player_stats_08_25_2019.csv
 create mode 100644 data/predictions/predictions_08_25_2019.csv
fatal: could not read Username for 'https://github.com': Device not configured
  0%|          | 0/15 [00:00<?, ?it/s]  7%|6         | 1/15 [01:00<14:05, 60.43s/it] 13%|#3        | 2/15 [02:03<13:15, 61.16s/it] 20%|##        | 3/15 [03:02<12:06, 60.51s/it] 27%|##6       | 4/15 [04:00<10:59, 59.92s/it] 33%|###3      | 5/15 [04:56<09:46, 58.66s/it] 40%|####      | 6/15 [05:57<08:54, 59.43s/it] 47%|####6     | 7/15 [08:25<11:27, 85.94s/it] 53%|#####3    | 8/15 [09:30<09:16, 79.52s/it] 60%|######    | 9/15 [10:29<07:21, 73.54s/it] 67%|######6   | 10/15 [11:32<05:52, 70.41s/it] 73%|#######3  | 11/15 [12:47<04:46, 71.60s/it] 80%|########  | 12/15 [13:47<03:24, 68.25s/it] 87%|########6 | 13/15 [14:47<02:11, 65.88s/it] 93%|#########3| 14/15 [15:58<01:07, 67.29s/it]100%|##########| 15/15 [17:08<00:00, 67.99s/it]
  0%|          | 0/8 [00:00<?, ?it/s] 12%|#2        | 1/8 [01:19<09:13, 79.12s/it] 25%|##5       | 2/8 [02:00<06:46, 67.72s/it] 38%|###7      | 3/8 [04:29<07:40, 92.04s/it] 50%|#####     | 4/8 [05:13<05:11, 77.83s/it] 62%|######2   | 5/8 [06:15<03:38, 72.98s/it] 75%|#######5  | 6/8 [07:03<02:11, 65.53s/it] 88%|########7 | 7/8 [08:18<01:08, 68.35s/it]100%|##########| 8/8 [09:25<00:00, 67.93s/it]
Finished generating file: data/player_stats/player_stats_08_25_2019.csv
Finished generating file: data/player_stats/player_stats_08_26_2019.csv
Results for 08/25/2019 generated
Getting data...
Data retrieved.
Training logistic regression...
Finished training!
Finding best hyperparameters for AdaBoost...
Training AdaBoost...
Finished training!
Finding best hyperparameters for random forests...
Training random forests...
Finished training!
Model performance summary on validation set: 
             Model  Accuracy  Precision    Recall  F1 Score
0          Logreg  0.733189   0.783505  0.652361  0.711944
1  Random Forests  0.731020   0.771144  0.665236  0.714286
2        AdaBoost  0.731020   0.765854  0.673820  0.716895
Predictions for today: 
                Name                  Team  Hit Probability
0     Marcus Semien     Oakland Athletics         0.801010
1       Gio Urshela      New York Yankees         0.800179
2   Freddie Freeman        Atlanta Braves         0.798702
3   Whit Merrifield    Kansas City Royals         0.794181
4  Ronald Acuna Jr.        Atlanta Braves         0.793338
5      Ozzie Albies        Atlanta Braves         0.791960
6      Tony Wolters      Colorado Rockies         0.790824
7        Mark Canha     Oakland Athletics         0.790652
8       Aaron Judge      New York Yankees         0.787353
9  Brandon Crawford  San Francisco Giants         0.784902
[master 08b03b8] predictions 08/26/19
 Committer: Rohan Narain <rohannarain@rohans-mbp-3.attlocal.net>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly. Run the
following command and follow the instructions in your editor to edit
your configuration file:

    git config --global --edit

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 6 files changed, 621 insertions(+), 338 deletions(-)
 create mode 100644 data/model_stats/performance_08_26_2019.csv
 create mode 100644 data/past_results/past_results_08_25_2019.csv
 rewrite data/player_stats/player_stats_08_25_2019.csv (88%)
 create mode 100644 data/player_stats/player_stats_08_26_2019.csv
 create mode 100644 data/predictions/predictions_08_26_2019.csv
fatal: could not read Username for 'https://github.com': Device not configured
  0%|          | 0/8 [00:00<?, ?it/s] 12%|#2        | 1/8 [01:21<09:32, 81.74s/it] 25%|##5       | 2/8 [02:32<07:49, 78.31s/it] 38%|###7      | 3/8 [03:38<06:13, 74.62s/it] 50%|#####     | 4/8 [04:44<04:48, 72.17s/it] 62%|######2   | 5/8 [05:39<03:21, 67.08s/it] 75%|#######5  | 6/8 [06:41<02:10, 65.46s/it] 88%|########7 | 7/8 [07:36<01:02, 62.24s/it]100%|##########| 8/8 [08:30<00:00, 59.92s/it]
  0%|          | 0/15 [00:00<?, ?it/s]  7%|6         | 1/15 [00:38<08:58, 38.45s/it] 13%|#3        | 2/15 [01:25<08:53, 41.04s/it] 20%|##        | 3/15 [02:06<08:12, 41.08s/it] 27%|##6       | 4/15 [03:05<08:31, 46.53s/it] 33%|###3      | 5/15 [04:07<08:30, 51.07s/it] 40%|####      | 6/15 [05:03<07:52, 52.51s/it] 47%|####6     | 7/15 [06:03<07:18, 54.76s/it] 53%|#####3    | 8/15 [07:48<08:08, 69.78s/it] 60%|######    | 9/15 [10:19<09:25, 94.26s/it] 67%|######6   | 10/15 [11:35<07:23, 88.72s/it] 73%|#######3  | 11/15 [12:20<05:02, 75.62s/it] 80%|########  | 12/15 [13:08<03:21, 67.21s/it] 87%|########6 | 13/15 [14:00<02:05, 62.70s/it] 93%|#########3| 14/15 [15:14<01:06, 66.29s/it]100%|##########| 15/15 [16:14<00:00, 64.16s/it]
Finished generating file: data/player_stats/player_stats_08_26_2019.csv
Finished generating file: data/player_stats/player_stats_08_27_2019.csv
Results for 08/26/2019 generated
